test/common/libs/shops.js

```diff
@@ -0,0 +1,72 @@
+import shared from '../../../common';
+import {
+  generateUser,
+} from '../../helpers/common.helper';
+
+describe('shops', () => {
+  let user = generateUser();
+
+  describe('market', () => {
+    let shopCategories = shared.shops.getMarketCategories(user);
+    it('contains at least the 3 default categories', () => {
+      expect(shopCategories.length).to.be.greaterThan(2);
+    });
+    it('does not contain an empty category', () => {
+      _.each(shopCategories, (category) => {
+        expect(category.items.length).to.be.greaterThan(0);
+      });
+    });
+    it('items contain required fields', () => {
+      _.each(shopCategories, (category) => {
+        _.each(category.items, (item) => {
+          expect(item).to.have.keys(['key', 'text', 'notes', 'value', 'currency', 'locked', 'purchaseType', 'class']);
+        });
+      });
+    });
+  });
+  describe('questShop', () => {
+    let shopCategories = shared.shops.getQuestShopCategories(user);
+    it('does not contain an empty category', () => {
+      _.each(shopCategories, (category) => {
+        expect(category.items.length).to.be.greaterThan(0);
+      });
+    });
+    it('items contain required fields', () => {
+      _.each(shopCategories, (category) => {
+        _.each(category.items, (item) => {
+          expect(item).to.have.keys('key', 'text', 'notes', 'value', 'currency', 'locked', 'purchaseType', 'boss', 'class', 'collect', 'drop', 'unlockCondition', 'lvl');
+        });
+      });
+    });
+  });
+  describe('timeTravelers', () => {
```

Comment:
These tests should cover the logic for what happens when a user owns a mystery set, pet or mount.